onnxruntime/core/providers/cpu/nn/tfidfvectorizer.cc

```diff
@@ -455,28 +455,33 @@ Status TfIdfVectorizer::ComputeImpl(OpKernelContext* ctx) const {
     C = input_dims[0];
     if (C < 1) {
       return Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT,
-                    "Input shape must have either [C] or [B,C] dimensions where C > 0 and B > 0");
+                    "Input shape must have either [C] or [B,C] dimensions where C >= 0 and B > 0");
     }
   } else if (input_dims.size() == 2) {
     B = input_dims[0];
     C = input_dims[1];
     b_dim = B;
-    if (B < 1 || C < 1) {
+    if (B < 1 || C < 0) {
       return Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT,
-                    "Input shape must have either [C] or [B,C] dimensions where C > 0 and B > 0");
+                    "Input shape must have either [C] or [B,C] dimensions where C >= 0 and B > 0");
     }
   } else {
     return Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT,
                   "Input shape must have either [C] or [B,C] dimensions where C > 0 and B > 0");
   }
 
-  assert((b_dim * C) == total_items);
-
   // Frequency holder allocate [B..output_size_]
   // and init all to zero
   std::vector<uint32_t> frequencies;
   frequencies.resize(b_dim * impl.output_size_, 0);
 
+  if (input_shape.Size() == 0) {
+    OutputResult(ctx, B, frequencies);
+    return Status::OK();
+  }
```

Comment:
Let's move this section UP before the frequencies and output the desired shape w/o using OutputResult(). OutputResult() is for processing a non-empty input which may result in empty output.